packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts

```diff
@@ -643,40 +644,51 @@ async function calculateUserPerformance(onChainProposal: ProposalAccountWithKey)
         .execute()
     })
 
-    let actors = <any>{}
-    
-    orders.forEach(o => {
-      const actor = o.actorAcct
-      if (!actors[actor]) {
-        actors[actor] = {
+    let actors = orders.reduce((current, next) => {
+      const actor = next.actorAcct
+      if (!current.get(actor)) {
+        current.set(actor, {
           tokensBought: 0,
           tokensSold: 0,
           volumeBought: 0,
           volumeSold: 0
-        }
+        })
+      }
+
+      
+      const tokenDecimals = tokens?.decimals
+      if (!tokenDecimals) {
+        // do we want to throw different type of error here?
```

Comment:
If no token decimals just dont set this entry in the map. You can just return the accumulated map here.